share/wake/lib/system/job.wake

```diff
@@ -114,6 +114,31 @@ export def setPlanOnce  v p = editPlanOnce  (\_ v) p
 export def setPlanKeep  v p = editPlanKeep  (\_ v) p
 export def setPlanShare v p = editPlanShare (\_ v) p
 
+# Create a labeled plan.
+# 'command' is a String, not List String.
+export def makeLabeledShellPlan label visible command =
+  makeShellPlan command visible
+  | setPlanLabel label
+
+export def makePlan2 = makeLabeledShellPlan
+
+# Prepend 'value' to the Plan's 'PATH' environment value
+export def prependPlanPath value plan =
+  editPlanEnvironment (addEnvironmentPath value) plan
+
+# Set an environment variable in a Plan
+export def setPlanEnvVar name value plan =
+  def var = "{name}={value}"
+  editPlanEnvironment (var, _) plan
+
+# Set the Job output to a single file
+export def setPlanOutput output plan =
+  setPlanOutputs (output, Nil) plan
```

Comment:
Hmm. I don't really want to encourage people to set the outputs of a Plan directly. I much prefer to guide people towards having wake to auto-detect them. Using 'obscurity' to guide people towards a different path might not be the best idea either, though.

Reply:
Perhaps a better helper function would be:
```
def filterPlanOutputs regExp plan = editPlanOutputFn (filter (matches regExp)) plan
```

Then your example would be: `| filterPlanOutputs `a``

Reply:
It's not exacally encouragement, it's a helper function in a source rather rather than something in a beginners how-to guide.

> | filterPlanOutputs `a` 

Would that work with setPlanLocalOnly?
I'm not actually looking to "filter" anything so I wouldn't have grep-ed for that

Reply:
It has backtics because it's a regular expression.

This proposal would NOT work with LocalOnly, which was kind of the point. ;-) I don't want to encourage LocalOnly.

Actually, what I think we might want is to change `setPlanLocalOnly` to accept the "output list" as its argument, rather than taking True/False.

Reply:
So, LocalOnly is its own thing. If you're using that, you are a power-user and you're using dangerous APIs. We could/should still make it harder to mess up, so I think the `setPlanLocalOnly` accepting the output list helps there.

The risk with setPlanOutput vs. filterPlanOutput is you can specify a file that does not actually get created by the Job. That introduces a new failure mode, that is not very easy to handle gracefully. For LocalOnly, we have to take that risk, but for non-LocalOnly, I would rather steer people away from the hole.